src/mlpack/methods/neighbor_search/neighbor_search.hpp

```diff
@@ -309,14 +412,21 @@ class NeighborSearch
   //! Return the number of node combination scores during the last search.
   size_t Scores() const { return scores; }
 
+  //! Access the search mode.
+  NeighborSearchMode SearchMode() const { return searchMode; }
+  //! Modify the search mode.
+  void SetSearchMode(const NeighborSearchMode mode);
```

Comment:
@rcurtin  Yes, I agree that it is preferable to have: `NeighborSearchMode& SearchMode()`
Suppose I add that method, the problem it this:
Until mlpack 3.0.0, we have the modifiers: `Naive()` and `SingleMode()`.
So, when a method is called, and I see a difference between the booleans values: `naive` and `singleMode` and the member `searchMode` (for example: `naive == true` , `singleMode == false` , `searchMode == DUAL_TREE_MODE` ) we don't know which fields were modified, the boolean flags `naive` and `singleMode`, or the `searchMode` member.  So, I don't know if the user want to do naive search or dual tree search.
Because of that I decided to include a  member `SetSearchMode()`.
One solution I can think, I am not sure if this is what you suggested, is to include the modifier `SearchMode()` and a comment like:
"Untile mlpack 3.0.0, if you modify the SearchMode, you also have to modify the flags Naive() and SingleMode()"
What do you think?
Thanks!